utilities/api/index.ts

```diff
@@ -0,0 +1,60 @@
+import { CONTENTFUL_API_ACCESS_TOKEN, CONTENTFUL_API_URL, HttpRequestMethod } from './constants'
+import type { PageContentProps } from '../../components/Main/PageContent'
+
+type CollectionData = { pageContentCollection: { items: Pathname[] } }
+type ItemData = { pageContentCollection: { items: [PageContentProps] } }
+type Data = CollectionData | ItemData
+export type Pathname = { pathname: string }
+
+async function fetchGraphQl(operationName: string, query: string): Promise<Data> {
+  const res = await fetch(CONTENTFUL_API_URL, {
+    method: HttpRequestMethod.POST,
+    headers: {
+      Authorization: CONTENTFUL_API_ACCESS_TOKEN,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      operationName,
+      query,
+    }),
+  })
+
+  const { data }: { data: Data } = await res.json()
+
+  return data
+}
+
+export async function getAllPathnames(): Promise<Pathname[] | []> {
+  const operationName = 'GetAllPathnames'
+  const query = `
+    query ${operationName} {
+      pageContentCollection {
+        items {
+          pathname
+        }
+      }
+    }
+  `
+
+  const data: Data = (await fetchGraphQl(operationName, query)) as CollectionData
+
+  return data.pageContentCollection.items
+}
+
```

Comment:
What if we did this instead?

```typescript
export type Path = { pathname: string }

const headers = {
  Authorization: CONTENTFUL_API_ACCESS_TOKEN,
  'Content-Type': 'application/json',
}

export async function listPagePaths(): Promise<Path[]> {
  const body = JSON.stringify({
    operationName: 'ListPagePaths',
    query: `query ListPagePaths {
      pageContentCollection {
        items {
          pathname
        }
      }
    }`,
  })

  const result = await fetch(CONTENTFUL_API_URL, {
    method: HttpRequestMethod.POST,
    headers,
    body,
  })

  const json = await result.json()

  return json?.data?.pageContentCollection?.items || []
}
```

This is a first pass at it, but some thoughts:

- `{ pathname: string }` is an _object_ with a pathname property, not the pathname, so the type should not be `Pathname` as that's very confusing (it confused me repeatedly).
- The headers are always the same, so set them outside the functions
- Generate the body as a JSON string separately to reduce noise. Now we can see what we're doing.
- Now the call to `fetch` is very simple and clear.
- We don't need to split the json out only to end up digging down into `data` anyway.
- Let's return an empty array (or should we throw?) if anything is missing.

I think that we should use `list` when we are returning all objects (or a filtered set) and `get` for single objects. That makes it very clear what we're doing. So `listPagePaths` means list all the page paths.